internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,637 @@
+		}
+
+		if dbInstallations.Contains(d.installationID) {
+			unlocked, err := d.acquireLock(databaseCluster.ID, store)
+			if err != nil {
+				return nil, nil, errors.Wrapf(err, "could not lock database installation id %", d.installationID)
+			}
+			defer unlocked(logger)
+
+			rdsCluster, err = d.describeRDSCluster(databaseCluster.ID)
+			if err != nil {
+				return nil, nil, errors.Wrap(err, "could not describe RDS cluster")
+			}
+
+			break
+		}
+	}
+
+	if rdsCluster == nil {
+		return nil, nil, errors.Errorf("could not find an RDS endpoint for installation id %s", d.installationID)
+	}
+
+	installationSecreteName := RDSMultitenantSecretName(d.installationID)
+
+	result, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: &installationSecreteName,
+	})
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to generate databa spec and secret")
+	}
+
+	installationSecret, err := unmarshalSecretPayload(*result.SecretString)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to generate databa spec and secret")
+	}
+
+	installationDBName := MattermostRDSDatabaseName(d.installationID)
+	installationDBConnString := MattermostMySQLConnString(installationDBName, *rdsCluster.Endpoint, installationSecret.MasterUsername, installationSecret.MasterPassword)
+
+	databaseSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: installationSecreteName,
+		},
+		StringData: map[string]string{
+			"DB_CONNECTION_STRING": installationDBConnString,
+		},
+	}
+
+	databaseSpec := &mmv1alpha1.Database{
+		Secret: installationSecreteName,
+	}
+
+	logger.Infof("Finish to set up spec and secret for database %s", installationDBName)
+	logger.Debugf("RDS DB cluster %s configured to use connection string %s", *rdsCluster.DBClusterIdentifier, installationDBConnString)
+
+	return databaseSpec, databaseSecret, nil
+}
+
+// Provision claims a multitenant RDS cluster and creates a database schema for the installation.
+func (d *RDSMultitenantDatabase) Provision(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) error {
+	databaseName := MattermostRDSDatabaseName(d.installationID)
+	logger = logger.WithField("multitenant-rds-database", databaseName)
+
+	vpc, err := d.getClusterInstallationVPC(store)
+	if err != nil {
+		return errors.Wrapf(err, "unable to find the VPC for installation ID %s", d.installationID)
+	}
+
+	lockedCluster, err := d.lockRDSCluster(*vpc.VpcId, store, logger)
+	if err != nil {
+		return errors.Wrapf(err, "unable to lock a RDS cluster for installation ID %s", d.installationID)
+	}
+	defer lockedCluster.unlock(logger)
+
+	masterSecretValue, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: lockedCluster.cluster.DBClusterIdentifier,
+	})
+	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
```

Comment:
I think this error check might not be right. The check says if err is not a resource not found exception.. then log "unable to find ..."

I'm a bit confused. If we're unable to find the secret, wouldn't we get a ResourceNotFoundException? What is the error case where this `if` evaluates to `true`?

Reply:
Nice catch, this is a bug 👍